joinmarket/support.py

```diff
@@ -398,3 +410,61 @@ def debug_dump_object(obj, skip_fields=None):
             log.debug(pprint.pformat(v))
         else:
             log.debug(str(v))
+    
+def save_session_to_file (tumbler,wallet_file,destaddrs,filename):
+    if not os.path.exists(os.path.dirname(filename)):
+        try:
+            os.makedirs(os.path.dirname(filename))
+        except OSError as exc:
+            if exc.errno != errno.EEXIST:
+                raise
+    f = open(filename,"w")
+    f.write('{"options":')
+    f.write(json.dumps(tumbler.options)+',\n')
+    f.write('"wallet": "'+wallet_file+'",\n')
+    f.write('"destaddrs":')
+    f.write(json.dumps(destaddrs)+',\n')
+    f.write('"tx_list":')
+    f.write(json.dumps(tumbler.tx_list)+',\n')
+    f.write('"next_tx": 0,\n')
+    f.write('"pushed_tx": "",\n')
+    f.write('"pushed_destaddr": ""}\n')
+    f.close()
+
+def update_session_tx_confirmed (sessionname,i_tx):
+    sessionfilename = os.path.join(script_dir,'../sessions/'+sessionname)
```

Comment:
Crashed here on a first test run due to sessionname not being set (it's 'None'). The help message says it's optional, so I guess can be fixed various ways, e.g. don't call this fn if sessionname option not chosen, or something similar. It's currently called unconditionally in TumblerThread.confirm_callback()